### ISSUE TITLE
Drop `pygeos`

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/dask-geopandas-feedstock#12]
Recently the bot update `0.3.0` PR ( https://github.com/conda-forge/dask-geopandas-feedstock/pull/11 ) was merged. However it did not include dependency updates. Upstream [`pygeos` was dropped]( https://github.com/geopandas/dask-geopandas/compare/v0.2.0...v0.3.0#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7L17 ). However it is still listed as a dependency here

https://github.com/conda-forge/dask-geopandas-feedstock/blob/7663128f7c8f9affb79e3e81780cafefe8560542/recipe/meta.yaml#L24

It would be good to drop the dependency in the recipe. May also be worthwhile to patch the repodata to drop the dependency. Also it may be worth enabling [Grayskull for the update bot]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#how-does-regro-cf-autotick-bot-create-automatic-version-updates ) by [updating `conda-forge.yml`]( https://conda-forge.org/docs/maintainer/conda_forge_yml.html#bot ). For example ( https://github.com/conda-forge/dask-cuda-feedstock/pull/16 )

**Comment by jorisvandenbossche** [comment]
Indeed, we should drop pygeos here too.

> May also be worthwhile to patch the repodata to drop the dependency

I suppose that is not needed (or at least not critical). 

Basically, the dependency logic that we would like to express is "shapely>=2 OR pygeos", but I don't think you can express such conditional dependency. 
So we removed `pygeos` as dependency to allow people to _not_ install it (if they know they have shapely>=2), but if you are using older shapely (not all packages already are compatible with the latest shapely), you still need pygeos.

**Comment by scottyhq** [comment]
Just came across this, as currently installing dask-geopandas from conda-forge brings in `pygeos` no matter what, even if you do have shapely2 and therefore the warning keeps coming on import (which is avoidable of course with the environment variable).

```python
UserWarning: Shapely 2.0 is installed, but because PyGEOS is also installed, GeoPandas will still use PyGEOS by default for now. To force to use and test Shapely 2.0, you have to set the environment variable USE_PYGEOS=0. You can do this before starting the Python process, or in your code before importing geopandas:

import os
os.environ['USE_PYGEOS'] = '0'
import geopandas

In a future release, GeoPandas will switch to using Shapely by default. If you are using PyGEOS directly (calling PyGEOS functions on geometries from GeoPandas), this will then stop working and you are encouraged to migrate from PyGEOS to Shapely 2.0 (https://shapely.readthedocs.io/en/latest/migration_pygeos.html).
  import geopandas
```

**Comment by weiji14** [comment]
Started a new PR at #15 to drop `pygeos` (based on #13, but without the `inspect-grayskull` bot stuff). Note that this is applied on dask-geopandas 0.3.1. Would need a repodata-patch for dask-geopandas 0.3.0 if there's interest.

**Comment by jorisvandenbossche** [comment]
Yes, we can probably just drop pygeos completely nowadays, and not try to express "shapely>=2 OR pygeos" (given that shapely 2.0 is no longer just released). @weiji14 Thanks for the PR!
